### PR TITLE
Unbreak support for globals in SQL over native protocol

### PR DESCRIPTION
**Pull request by elprans**

### CODE REVIEW

File: edb/server/compiler/compiler.py [hunk 1]
```diff
@@ -2486,10 +2486,12 @@ def compile_sql_as_unit_group(
...
 
-    apply_access_policies_sql = None
-    setting = _get_config_val(ctx, 'apply_access_policies_sql')
+    # Note that unlike SQL over PostgreSQL protocol we use
+    # the general access policy toggle, not the SQL-specific one.
+    apply_access_policies = None
+    setting = _get_config_val(ctx, 'apply_access_policies')
     if setting:
-        apply_access_policies_sql = sql.is_setting_truthy(setting)
+        apply_access_policies = sql.is_setting_truthy(setting)
```
[the remark this answers is not in the document]
Comment (elprans) [reply edb/server/compiler/compiler.py:2494]: PG protocol is a completely distinct use case, we don't expect people to be mixing and matching it with normal bindings.  Basically, PG protocol is for external tooling, ETL, egress etc where access policies would actually likely be _detrimental_ unless you're very deliberate in setting them in a way that can be easily toggled with a global or something.

On the other hand, SQL over edb proto is intended to be used the same way EdgeQL is -- in apps, possibly mixing `query` and `query_sql` freely, in which case using normal access policy logic is expected.

> If we go with this option, I'd rename config `apply_access_policies_sql` to `apply_access_policies_pg`, so it signifies that it refers to the "postgres protocol", not "sql as the language".

I'm not quite happy with the naming either and also `apply_access_policies` should probably have been an enum in the first place (we have a rule of not adding boolean configs).

[the remark this answers is not in the document]
Comment (aljazerzen) [reply edb/server/compiler/compiler.py:2494]: > PG protocol is for external tooling

Yes, generally, but there are also ORMs or database clients (such as DataGrip / DBeaver) that would use the PG protocol. If a developer would be using `.query_sql` along with their "database UI" tool of choice, they would be seeing different results.

**Review by msullivan — APPROVED** [review]
We can argue more on the call though if we want

File: edb/server/compiler/compiler.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (aljazerzen) [reply edb/server/compiler/compiler.py:2494]: We decided that sql-over-edgedb-protocol should use `apply_access_policies` and sql-over-pg-protocol should use `apply_access_policies_pg`

**Pull request merged**